### DISCUSSION TITLE
setup webpack config in next.config.js

### DISCUSSION THREAD

**Discussion by ahmadbaihaqi** [vercel/next.js#20188] [Help]
I'm working with NextJS and using [react-data-export][1] plugin to generate xls files.
in the description it says :

> This library uses [file-saver][2] and [xlsx][3] and using
> [json-loader][4] will do the magic for you.
> 
>     ///webpack.config.js
>     vendor: [
>             .....
>             'xlsx',
>             'file-saver'
>     ],
>     .....
>     node: {fs: 'empty'},
>     externals: [
>         {'./cptable': 'var cptable'},
>         {'./jszip': 'jszip'}
>      ]

but I have no idea how to implement it and got error like this :

    The static directory has been deprecated in favor of the public directory. https://err.sh/vercel/next.js/static-dir-deprecated
    Defining routes from exportPathMap
    event - compiled successfully
    > Ready on http://localhost:80 or http://localhost 
    > Ready on https://localhost:443 or https://localhost 
    event - build page: /menu_accounting/ReportGross
    wait  - compiling...
    error - ./node_modules/react-export-excel/node_modules/xlsx/xlsx.js
    Module not found: Can't resolve 'fs' in '/home/ahmadb/repos/lorry-erp/node_modules/react-export-excel/node_modules/xlsx'
    Could not find files for /menu_accounting/ReportGross in .next/build-manifest.json

    

  [1]: https://www.npmjs.com/package/react-data-export
  [2]: https://www.npmjs.com/package/file-saver
  [3]: https://www.npmjs.com/package/xlsx
  [4]: https://www.npmjs.com/package/json-loader

**Comment by Carlosf96** [comment]
Has anyone been able to solve this?

**Comment by behnam-io** [comment]
I also have the same issue, can anyone help with this?

**Reply by Carlosf96** [reply]
Helloo yes I was able to solve it using a different library could you try using this library ? "xlsx": "^0.17.0"
